physics/cires_vert_orodis_v1.F90

```diff
@@ -0,0 +1,1037 @@
+! version 3 => ncep/gfs-2017 -gfs_2017 with lm1997
+!======================== 
+!     real, parameter   ::  Fcrit_mtb  = 0.7   
+      
+      integer, intent(in)              ::  nz     
+      integer, intent(in)              ::  iemax    ! standard ktop  z=elvpd  + 4 * hprime
+      real  , intent(out)              ::  taumtb    
+          
+      integer , intent(out)            ::  idxzb           
+      real, dimension(nz), intent(out) ::  drmtb
+
+      real, intent(in)  ::  elvp, elvpd              !elvp  = min (elvpd  + sigfac * hprime(j), hncrit=10000meters)
+      real, intent(in)  ::  hprime , sigma, theta, oc,  oa4(4), clx4(4), gam
+      real, intent(in)  ::  zpbl     
+
+      real, dimension(nz),    intent(in) :: up, vp, tp, qp, dp, zpm, pmid
+      real, dimension(nz+1),  intent(in) :: zpi, pint
+!
+      real, dimension(nz+1)              :: zpi_zero
+      real, dimension(nz)                :: zpm_zero
+      real ::  vtj, rhok, bnv2, rdz, vtkp, vtk, dzp
+      
+      real, dimension(nz) ::  bn2, uds, umf, cosang, sinang
+      
+      integer :: k,  klow, ktop, kpbl
+      real    :: uhm, vhm, bn2hm, rhohm,                  &
+                 mtb_fix, umag, bnmag, frd_src,           &
+                 zblk, who_iz_normal, rlm97,              &
+                 phiang, ang, pe, ek,                     &
+                 cang, sang, ss2, cs2, zlen,  dbtmp,      &
+                 hamp, bgamm, cgamm      
+  
+!==================================================
+!
+!     elvp + hprime <=>elvp + nridge*hprime, ns =2
+!                 ns = sigfac
+!     tau_parel & tau_normal along major "axes" 
+!
+!  options to block the "flow", choices for [klow, ktop]
+!
+!  1-directional (normal)  & 2-directional "blocking"
+!
+!==================================================
+! no - blocking: drmtb(1:nz) = 0.0
+!=================
+      idxzb = -1
+      drmtb(1:nz) = 0.0
+      taumtb      = 0.0
+      klow = 2
+
+      ktop = iemax
+      hamp = nridge*hprime
+      
+! reminder:        cdmb = 4.0 * 192.0/float(imx)*cdmbgwd(1)   Lellipse= a/2=sigma/hprime
+
+      mtb_fix = cdmb*sigma/hamp        !hamp  ~ 2*hprime and 1/sigfac = 0.25 is inside 1/hamp
+
+      if (mtb_fix == 0.) then
+        print *, cdmb, sigma, hamp
+        print *,  ' MTB == 0'
+           stop
+      endif
+
+      if (strver == 'vay_2018')  then    
+
+        zpm_zero = zpm - zpi(1)
+        zpi_zero = zpi - zpi(1)
+
+         do k=1, nz-1
+           if (hamp .le. zpi_zero(k+1) .and. (hamp .gt. zpi_zero(k)  ) ) then
+               ktop  =  k+1      !......simply k+1 next interface level
+               exit
+            endif
+          enddo
+!        print *,  klow, ktop,   ' klow-ktop '
+        call um_flow(nz, klow, ktop, up, vp, tp, qp, dp, zpm, zpi, pmid, pint, &
+             bn2, uhm, vhm, bn2hm, rhohm)   
+     
+        umag = max(sqrt(uhm*uhm + vhm*vhm), velmin)       !velmin=dw2min =1.0 m/s 
+        if (bn2hm .le. 0.0) then
```

Comment:
Same. We should be using CCPP error-handling here.

Reply:
As in the above, I changed and commented out these lines:
        ! if (bn2hm .le. 0.0) then
        !    write(errmsg,'(*(a))') 'unstable MF for MTB  - RETURN '
        !    errflg = 1
        !    return
        ! end if

Will uncomment the lines later if we need to activate them.